managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -195,11 +195,11 @@
     private final static CompletableFuture<PositionImpl> NULL_OFFLOAD_PROMISE = CompletableFuture
             .completedFuture(PositionImpl.latest);
     private volatile LedgerHandle currentLedger;
-    private long currentLedgerEntries = 0;
-    private long currentLedgerSize = 0;
-    private long lastLedgerCreatedTimestamp = 0;
-    private long lastLedgerCreationFailureTimestamp = 0;
-    private long lastLedgerCreationInitiationTimestamp = 0;
+    private volatile long currentLedgerEntries = 0;
```

Comment:
All of these variable should be getting update while holding the mutex on the `ManagedLedgerImpl` instance, so we shouldn't anyway requiring atomicity here

Reply:
That's true for updating, but reading doesn't seem to hold the mutex in all cases? That could lead to "word tearing" problem at read time on 32-bit JVMs? As mentioned in the other comments, this isn't an issue on 64-bit JVMs. Perhaps running the Broker on non-64 bit JVMs should simply be unsupported and possibly prevented at startup?

Reply:
> Perhaps running the Broker on non-64 bit JVMs should simply be unsupported and possibly prevented at startup?

That's a good point. I think it would a good option to just preventing it to start. Right now we don't have any check that it actually works on 32bit. And, while it's true that there are still many pieces of hardware out there on 32bits CPUs, I'd say it's unlikely that you'd want to run brokers on them. Even small devices like RPi are running Arm64.